internal/common/teleport.go

```diff
@@ -0,0 +1,61 @@
+package common
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/cluster-test-suites/internal/state"
+	"github.com/giantswarm/cluster-test-suites/internal/teleport"
+	"github.com/giantswarm/clustertest/pkg/logger"
+	"github.com/giantswarm/clustertest/pkg/wait"
+
+	tc "github.com/gravitational/teleport/api/client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func runTeleport(teleportSupported bool) {
+	Context("teleport", func() {
+		var teleportClient *tc.Client
+
+		isClusterRegistered := func(clusterName string) (bool, error) {
```

Comment:
What's the point of this variable rather than just inlining the function in the `It` below?

Reply:
Hmm...I used it based on similar approach in `dns.go`
> https://github.com/giantswarm/cluster-test-suites/blob/d5285e05b9710e45af8f3d40db4374de68b41133/internal/common/dns.go#L24-L33

Reply:
But your function is only used in a single place. It's not shared across tests so no need to have it as a variable unless there's something I'm missing.

Reply:
Also, the way that Ginkgo runs code can be a little surprising sometimes so where possible you should avoid running login outside of an `It` or the `Before/After` hooks.

Reply:
Okay, gotcha.

Reply:
Addressed in commit https://github.com/giantswarm/cluster-test-suites/pull/223/commits/262d74f4fabca2a41614ae82d2bf58a386641c0e